### PR TITLE
Feature/rb 17

### PR DESCRIPTION
**Pull request by Saverio976**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [x] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

https://x-r-g-b.atlassian.net/browse/RB-17?atlOrigin=eyJpIjoiNDBmMDM5ZWI4ODU1NGQ5YmI4ODc3ZTRkYzJjMjc2ODEiLCJwIjoiaiJ9

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: src/Client/Systems/ClientSystems.hpp [hunk 1]
```diff
@@ -10,9 +10,9 @@
 #include <cstddef>
 
 namespace GraphicSystems {
-    void pixelRenderer(std::size_t);
+    void pixelRenderer(std::size_t /*unused*/);
```
Comment (TTENSHII) [inline src/Client/Systems/ClientSystems.hpp:13]:
```suggestion
    void pixelRenderer(std::size_t UNUSED);
```

File: src/Client/Systems/ClientSystems.hpp [hunk 2]
```diff
@@ -10,9 +10,9 @@
...
 }
 
 namespace EventsSystems {
-    void playerMovement(std::size_t);
+    void playerMovement(std::size_t /*unused*/);
```
Comment (TTENSHII) [inline src/Client/Systems/ClientSystems.hpp:17]:
```suggestion
    void playerMovement(std::size_t UNUSED);
```

File: src/ECS/Systems/Systems.cpp [hunk 3]
```diff
@@ -5,28 +5,35 @@
...
 */
 
+#include <cstddef>
 #include <iostream>
 #include "CustomTypes.hpp"
 #include "Registry.hpp"
 
 namespace Systems {
-    void collision(std::size_t)
+    void collision(std::size_t /*unused*/)
```
Comment (TTENSHII) [inline src/ECS/Systems/Systems.cpp:14]:
```suggestion
    void collision(std::size_t UNUSED)
```

File: src/Client/Systems/ClientSystems.cpp [hunk 4]
```diff
@@ -11,23 +11,25 @@
...
+        for (int i = 0; i < pixelRenderNumber; i++) {
+            for (int j = 0; j < pixelRenderNumber; j++) {
+                DrawPixel(begin.value().x + i, begin.value().y + j, PURPLE);
             }
         }
     }
 }
 
-void EventsSystems::playerMovement(std::size_t)
+void EventsSystems::playerMovement(std::size_t /*unused*/)
```
Comment (TTENSHII) [inline src/Client/Systems/ClientSystems.cpp:32]:
```suggestion
void EventsSystems::playerMovement(std::size_t UNUSED)
```

File: src/Client/Systems/ClientSystems.cpp [hunk 5]
```diff
@@ -11,23 +11,25 @@
 #include "CustomTypes.hpp"
 #include "Registry.hpp"
 
-void GraphicSystems::pixelRenderer(std::size_t)
+static const auto pixelRenderNumber = 50;
+
+void GraphicSystems::pixelRenderer(std::size_t /*unused*/)
```
Comment (TTENSHII) [inline src/Client/Systems/ClientSystems.cpp:16]:
```suggestion
void GraphicSystems::pixelRenderer(std::size_t UNUSED)
```

**Review by kw4ntiq — APPROVED** [review]

**Review by kw4ntiq — APPROVED** [review]

File: src/ECS/EventManager/EventManager.cpp [hunk 6]
```diff
@@ -26,12 +29,10 @@ void EventManager::updateEvents()
     }
 }
 
-bool EventManager::checkEvent(int event)
+auto EventManager::checkEvent(int event) -> bool
```
Comment (guillaumeAbel) [inline src/ECS/EventManager/EventManager.cpp]: jamais

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

File: src/ECS/EventManager/EventManager.cpp (hunk 6 above)
Comment (Saverio976) [reply src/ECS/EventManager/EventManager.cpp]: https://stackoverflow.com/questions/52103775/advantage-of-using-trailing-return-type-in-c11-functions

Comment (Saverio976) [reply src/ECS/EventManager/EventManager.cpp]: https://danielsieger.com/blog/2022/01/28/cpp-trailing-return-types.html

**Pull request merged**